### PR TITLE
    Added minor changes to utils/passwordresetting and ForgetPassword…

### PR DESCRIPTION
**Pull request by Byte4Craft**
… method in views.py

    Corrected docstrings in json_loader method
    Removed commented internationalization and password validations strings from settings.py

## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVPYTP-34)

## Code reviewers

- [ ] @dmytrorevak
- [ ] @IvanAndrukh
- [ ] @JjuliaP
- [ ] @makarman
- [ ] @noobhunters
- [ ] @Silent777
- [ ] @Staryj812

### Second Level Review

- [ ] @ssOleg
- [ ] @lhalam

## Summary of issue

ToDo

## Summary of change

Added minor changes to utils/passwordresetting and ForgetPassword method in urls.py
Corrected docstrings in json_loader method
Removed commented internationalization and password validations strings from settings.py

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: eventually/authentication/views.py [hunk 1]
```diff
@@ -5,16 +5,109 @@
...
 
 class UserView(View):
-    """UserView view handles GET, POST, PUT, DELETE requests."""
+    """    A class to handle GET, PUT and DELETE operations    """
+
+    def get(self, request, user_id):
+        """
+        Handles GET request, retrieves a user profile from the database
+        :param request: request from the web page
+        :type request: WSGIRequest
```
Comment (redmyt) [inline eventually/authentication/views.py]: this is a simple HTTP request.

File: eventually/authentication/views.py [hunk 2]
```diff
@@ -5,16 +5,109 @@
...
+                new_password = new_attrs['password']
+            else:
+                return HttpResponse("Password doesn't meet the requirements", status=400)
+
+            if new_attrs.get('email') and email_validator(new_attrs['email']):
+                new_email = new_attrs['email']
+            else:
+                return HttpResponse("You did not provide a valid email", status=400)
+
+            try:
```
Comment (redmyt) [inline eventually/authentication/views.py]: is this `try` block necessary?

File: eventually/authentication/views.py [hunk 3]
```diff
@@ -5,16 +5,109 @@
...
+        if new_attrs:
+            user_to_modify = CustomUser.get_by_id(user_id)
+
+            if new_attrs.get('password') and password_validator(new_attrs['password']):
+                new_password = new_attrs['password']
+            else:
+                return HttpResponse("Password doesn't meet the requirements", status=400)
+
+            if new_attrs.get('email') and email_validator(new_attrs['email']):
+                new_email = new_attrs['email']
```
Comment (redmyt) [inline eventually/authentication/views.py]: can we modify the user's email?

File: eventually/authentication/views.py [hunk 4]
```diff
@@ -5,16 +5,109 @@
...
+        :param user_id: id of a user which has to be changed
+        :type user_id: int
+
+        :return: status 200 if user has been updated,
+                 status 400 if user hasn't been updated,
+        """
+
+        new_attrs = json_loads(request.body)
+        if new_attrs:
+            user_to_modify = CustomUser.get_by_id(user_id)
```
Comment (redmyt) [inline eventually/authentication/views.py]: add verification is request.user is user_to_modify.

File: eventually/authentication/views.py [hunk 5]
```diff
@@ -5,16 +5,109 @@
...
+
+            if new_attrs.get('password') and password_validator(new_attrs['password']):
+                new_password = new_attrs['password']
```
Comment (redmyt) [inline eventually/authentication/views.py]: use `set_password`

File: eventually/authentication/views.py [hunk 6]
```diff
@@ -5,16 +5,109 @@
...
+            else:
+                return HttpResponse("Password doesn't meet the requirements", status=400)
```
Comment (redmyt) [inline eventually/authentication/views.py]: is password credential necessary?
can we change our password without old password verification?

File: eventually/authentication/views.py [hunk 7]
```diff
@@ -5,16 +5,109 @@
...
+        """
+        Handles delete request
+
+        :param user_id: id of a user to delete
+        :type user_id: int
+
+        :return: status 200
+        """
+
+        if CustomUser.delete_by_id(user_id):
```
Comment (redmyt) [inline eventually/authentication/views.py]: with that logic, I can simply delete your account =)

**Review by redmyt — CHANGES_REQUESTED** [review]

File: eventually/authentication/views.py (hunk 1 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: Changed back.

File: eventually/authentication/views.py (hunk 5 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: Not needed because set_password is in models.py

File: eventually/authentication/views.py (hunk 4 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: Fixed, my bad.

File: eventually/authentication/views.py (hunk 7 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: Now you can't.)

File: eventually/authentication/views.py (hunk 6 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: This is the new password which is being verified, not the old one.)

File: eventually/authentication/views.py (hunk 3 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: Ability to change user's email has been removed both from models and views.

File: eventually/authentication/views.py (hunk 2 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: No, not necessary. Removed it.

File: eventually/authentication/views.py (hunk 6 above)
Comment (Silent777) [reply eventually/authentication/views.py]: We only need statuses without messages. We put massages in UI probably :)

Comment (Byte4Craft) [reply eventually/authentication/views.py]: Corrected.

File: eventually/authentication/views.py [hunk 8]
```diff
@@ -20,7 +20,89 @@
...
+    def get(self, request, user_id):
+        """
+        Handles GET request, retrieves a user profile from the database
+        :param request: request from the web page
+        :type request: HttpRequest
+
+        :param user_id: id of a user to return.
+        :type user_id: int
+
+        :return: user dictionary
```
Comment (lhalam) [inline eventually/authentication/views.py]: incorrect description, only dict?

Comment (Byte4Craft) [reply eventually/authentication/views.py]: My bad, fixed.

File: eventually/authentication/views.py [hunk 9]
```diff
@@ -19,7 +19,95 @@
...
+
+        :return: status 200 if user has been updated,
+                 status 400 if user hasn't been updated,
+        """
+
+        new_attrs = request.body
+
+        user = CustomUser.get_by_id(user_id)
+        if not user:
+            return HttpResponse(status=400)
```
Comment (redmyt) [inline eventually/authentication/views.py]: status code 404

File: eventually/authentication/views.py [hunk 10]
```diff
@@ -19,7 +19,95 @@
...
+        if new_attrs and (user.id == request.user.id):
+            user_to_modify = CustomUser.get_by_id(user_id)
```
Comment (redmyt) [inline eventually/authentication/views.py]: unnecessary line, see line 74

File: eventually/authentication/views.py [hunk 11]
```diff
@@ -19,7 +19,95 @@
...
+        | }
+
+        :param user_id: id of a user which has to be changed
+        :type user_id: str
+
+        :return: status 200 if user has been updated,
+                 status 400 if user hasn't been updated,
+        """
+
+        new_attrs = request.body
```
Comment (redmyt) [inline eventually/authentication/views.py]: add validation for all fields

File: eventually/authentication/views.py [hunk 12]
```diff
@@ -19,7 +19,95 @@
...
+
+        :param user_id: id of a user to delete
+        :type user_id: str
+
+        :return: status 200
+        """
+
+        user = CustomUser.get_by_id(user_id)
+        if not user:
+            return HttpResponse(status=400)
```
Comment (redmyt) [inline eventually/authentication/views.py]: status code 404

File: eventually/authentication/views.py [hunk 13]
```diff
@@ -19,7 +19,95 @@
...
+        if user.id == request.user.id:
+            if CustomUser.delete_by_id(user_id):
+                return HttpResponse(status=200)
+        return HttpResponse(status=400)
```
Comment (redmyt) [inline eventually/authentication/views.py]: status code 403

File: eventually/authentication/views.py [hunk 14]
```diff
@@ -19,7 +19,95 @@
...
+        Handles GET request, retrieves a user profile from the database
+        :param request: request from the web page
+        :type request: HttpRequest
+
+        :param user_id: id of a user to return.
+        :type user_id: int
+
+        :return: User profile Json and Status 200 if user exists, otherwise Status 404
+            E.G.
+                {
```
Comment (redmyt) [inline eventually/authentication/views.py]: add `|`

Comment (Byte4Craft) [reply eventually/authentication/views.py]: Fixed.

File: eventually/authentication/views.py (hunk 11 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: Added

File: eventually/authentication/views.py (hunk 9 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: Fixed.

File: eventually/authentication/views.py (hunk 10 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: Fixed

File: eventually/authentication/views.py (hunk 12 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: fixed.

File: eventually/authentication/views.py (hunk 13 above)
Comment (Byte4Craft) [reply eventually/authentication/views.py]: Fixed.

**Review by lhalam — DISMISSED** [review]

**Review by skramar — DISMISSED** [review]

**Review by lhalam — DISMISSED** [review]

**Review by lhalam — APPROVED** [review]

**Review by redmyt — APPROVED** [review]

**Pull request merged**